Source/core/CyclicBuffer.cpp

```diff
@@ -506,14 +519,10 @@ namespace Core {
 
                 _administration->_agents++;
 
-                AdminUnlock();
-
-                timeLeft = SignalLock(timeLeft);
+                result = SignalLock(timeLeft);
```

Comment:
Removing the AdminUnlock here, just before the waiting for a new lock might easily lead to deadlocks.
Can we associate these  "lock" changes at lease with a test case that showed that it was failing?